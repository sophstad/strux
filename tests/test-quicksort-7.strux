int main() {
    string[5] a = ["kofi", "josh", "millie", "sophie", "edwards"];
    a.showQuickSort();
    return 0;
}