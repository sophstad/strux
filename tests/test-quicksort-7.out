=====================================================================================
At this step:
current array: [ kofi josh millie sophie edwards ]
pivot swapped: kofi,edwards
array after swap: [ edwards josh millie sophie kofi ]
pivot swapped: millie,kofi
array after swap: [ edwards josh kofi sophie millie ]
strings swapped: sophie,kofi
array after swap: [ edwards josh sophie kofi millie ]
pivot is kofi
strings swapped: sophie,kofi
array after swap: [ edwards josh kofi sophie millie ]
strings swapped: millie,sophie
array after swap: [ edwards josh kofi millie sophie ]
=====================================================================================
At this step:
current array: [ edwards josh kofi millie sophie ]
pivot is josh
=====================================================================================
At this step:
current array: [ edwards josh kofi millie sophie ]
pivot is edwards
=====================================================================================
QuickSort complete! Final Result: [ edwards josh kofi millie sophie ]
